documentation/library_guide/kernel_templates/single_pass_scan.rst

```diff
@@ -0,0 +1,208 @@
++===============================================+=====================================================================+
+|  ``q``                                        | The SYCL* queue where kernels are submitted.                        |
++-----------------------------------------------+---------------------------------------------------------------------+
+|                                               | The sequences to apply the algorithm to.                            |
+|  - ``in_begin``, ``in_end``, ``out_begin`` (1)| Supported sequence types:                                           |
+|  - ``in_rng``, ``out_rng`` (2)                |                                                                     |
+|                                               | - ``sycl::buffer`` (2),                                             |
+|                                               | - :ref:`oneapi::dpl::experimental::ranges::views::all               |
+|                                               |   <viewable-ranges>` (2),                                           |
+|                                               | - :ref:`oneapi::dpl::experimental::ranges::views::subrange          |
+|                                               |   <viewable-ranges>` (2),                                           |
+|                                               | - :ref:`USM pointers <use-usm>` (1),                                |
+|                                               | - :ref:`oneapi::dpl::begin and oneapi::dpl::end                     |
+|                                               |   <use-buffer-wrappers>` (1).                                       |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+|  ``binary_op``                                | A function object that is applied to the elements of the input.     |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+|  ``param``                                    | A :doc:`kernel_param <../kernel_configuration>` object.             |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+
+
+**Type Requirements**:
+
+- The element type of sequence to scan must be a C++ integral or floating-point type
+  other than ``bool`` with a width of up to 64 bits.
+- The result is non-deterministic if the binary operator is non-associative (such as in floating-point addition).
```

Comment:
I suppose there should be commutativity as well.

The implementation uses reduction group operation, which requires both commutativity and associativity:
https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#:~:text=The%20result%20of%20a%20call%20to%20these%20functions%20is%20non%2Ddeterministic%20if%20the%20binary%20operator%20is%20not%20commutative%20and%20associative.

Reply:
Thanks, added.